sourcegraph/sourcegraph
# cody web: do not include ignored files in context

Excludes files from Cody scope if they match patterns from the Cody ignore file.

**Ignored file content** 👇🏻 
<img width="1904" alt="Screenshot 2024-01-29 at 15 55 53" src="https://github.com/sourcegraph/sourcegraph/assets/25318659/c7b111f5-b007-4f65-a792-9877a24ce90b">


| Ignored | Chat | Transcript |
| -- | -- | -- |
| ✅  |<img width="1904" alt="Screenshot 2024-01-29 at 15 54 51" src="https://github.com/sourcegraph/sourcegraph/assets/25318659/203009c6-58a8-4ddc-9357-c67b27b010fb">|<img width="900" alt="Screenshot 2024-01-29 at 16 00 20" src="https://github.com/sourcegraph/sourcegraph/assets/25318659/8ee4ae56-4ff6-45dd-b892-7acc0d910238">|
| 🚫 |<img width="1904" alt="Screenshot 2024-01-29 at 15 55 25" src="https://github.com/sourcegraph/sourcegraph/assets/25318659/a7643423-db5c-4496-a185-87e5a48680ba">|<img width="900" alt="Screenshot 2024-01-29 at 15 57 15" src="https://github.com/sourcegraph/sourcegraph/assets/25318659/2d66dafb-9f97-41de-8114-bae415b22e1d">|

> [!note]
> Known issue: Cody chat refers to the previously opened file in the current transcript if the current file is excluded from the scope (https://github.com/sourcegraph/cody/issues/2925). FYI when testing this PR. 


## Test plan
- Open repo which has `.cody/ignore` file
- Open a file matching the pattern from the ignore file
- View chat transcript (e.g., network tab) and ensure that code snippets from the ignored file are not included
